SetupDataPkg/ConfDataSettingProvider/ConfDataSettingProvider.c

```diff
@@ -294,6 +294,179 @@ DFCI_SETTING_PROVIDER  mSettingsProvider = {
   (DFCI_SETTING_PROVIDER_SET_DEFAULT)ConfDataSetDefault
 };
 
+/**
+  Set new runtime value to variable storage.
+
+  @param This      Provider Setting
+  @param Value     a pointer to the variable list
+  @param ValueSize Size of the data for this setting.
+  @param Flags     Informational Flags passed to the SET and/or Returned as a result of the set
+
+  @retval EFI_SUCCESS If setting could be set.  Check flags for other info (reset required, etc)
+  @retval Error       Setting not set.
+**/
+EFI_STATUS
+EFIAPI
+RuntimeDataSet (
+  IN  CONST DFCI_SETTING_PROVIDER  *This,
+  IN        UINTN                  ValueSize,
+  IN  CONST UINT8                  *Value,
+  OUT DFCI_SETTING_FLAGS           *Flags
+  )
+{
+  EFI_STATUS  Status = EFI_SUCCESS;
+  CHAR16 *VarName = NULL;
+  CHAR16 *name;
+  EFI_GUID *Guid;
+  UINT32 Attributes;
+  CHAR8 *Data;
+  UINT32 CRC32;
+  UINT32 LenToCRC32;  
+  UINT32 CalcCRC32 = 0;
+  
+  if ((This == NULL) || (This->Id == NULL) || (Flags == NULL) || (Value == NULL)) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  *Flags = 0;
+
+  if (0 != AsciiStrnCmp (This->Id, DFCI_OEM_SETTING_ID__RUNTIME, DFCI_MAX_ID_LEN)) {
+    DEBUG ((DEBUG_ERROR, "RuntimeDataSet was called with incorrect Provider Id (%a)\n", This->Id));
+    return EFI_UNSUPPORTED;
+  }
+
+  for (int i = 0; i < ValueSize;) {
+    // index into variable list
+    RUNTIME_VAR_LIST_HDR *VarList = (RUNTIME_VAR_LIST_HDR *)(Value + i);
+    /*
+     * Var List is in DmpStore format:
+     * 
+     *  struct {
+     *    RUNTIME_VAR_LIST_HDR VarList;
+     *    char Name[VarList->NameSize];
+     *    char Guid[16];
+     *    UINT32 Attributes;
+     *    char Data[VarList->DataSize];
+     *    UINT32 CRC32; // CRC32 of all bytes from VarList to end of Data
+     *  }
+     */
+    name = (CHAR16 *)(VarList + 1);
+    Guid = (EFI_GUID *)((CHAR8 *)name + VarList->NameSize);
+    Attributes = *(UINT32 *)(Guid + 1);
+    Data = (CHAR8 *)Guid + sizeof(*Guid) + sizeof(Attributes);
+    CRC32 = *(UINT32 *)(Data + VarList->DataSize);
+    LenToCRC32 = sizeof(*VarList) + VarList->NameSize + VarList->DataSize + sizeof(*Guid) + sizeof(Attributes);  
+
+    // on next iteration, skip past this variable
+    i += LenToCRC32 + sizeof(CRC32);

```

Comment:
A check of this index vs. the total size before dereferencing the buffer?

Reply:
The NameSize and DataSize as input value could be `insane`.

Reply:
Good call, will add.